### PR TITLE
BG-18745 Implement initial Drone CI for this KRSv2 project.

### PR DESCRIPTION
**Pull request by james8nguyen**
## JIRA Ticket
[https://bitgoinc.atlassian.net/browse/BG-18745](https://bitgoinc.atlassian.net/browse/BG-18745)

## Change Description
The following changes have been added:
1) Add initial Drone CI pipeline with .drone.jsonnet (with Danny G's help converting from .drone.yml to .drone.jsonnet. Didn't need it.)

2) In "checks" pipeline, add "lint test" which shows quite a few errors which need to be fixed.
   
3) Add pipeline with "unit tests" for ```node 6```. There are failing tests to be fixed.
https://drone.bitgo-dev.com/BitGo/key-recovery-service-v2/20/2/5

4) Add pipeline with both "unit tests" and "audit test" for ```node 10```. 
This pipeline is failing for obviously reasons. To resolve the failures and vulnerabilities, both package.json and package-lock.json must be upgraded with later versions of dependencies, and below is the PR for that.
[https://github.com/BitGo/key-recovery-service-v2/pull/70](https://github.com/BitGo/key-recovery-service-v2/pull/70)

## Tests
* Drone CI pipeline: https://drone.bitgo-dev.com/BitGo/key-recovery-service-v2/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,11 +5,14 @@
   "main": "index.js",
   "scripts": {
     "test": "./node_modules/mocha/bin/_mocha --timeout 20000 --recursive test/",
-    "start": "node -r dotenv/config bin/server.js"
+    "start": "node -r dotenv/config bin/server.js",
+    "lint": "./node_modules/.bin/eslint --ext .js .",
+    "audit": "npm audit"
```
Comment (dhoffmann) [inline package.json]: `run npm audit` instead of just using `npm audit`?

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (tylerlevine) [inline .eslintrc.json]: Can this eslint change be extracted into a separate PR?

File: .drone.jsonnet [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+          "npm run test",
+        ],
+        environment = {
+          MONGO_URI: "mongodb://mongo:27017/key-recovery-service-test",
+        },
+      ),
+      step(
+        name = "audit",
+        image = "node:10",
+        failure = true,
```
Comment (tylerlevine) [inline .drone.jsonnet]: Does this mean fail on error or ignore error?

File: package.json (hunk 1 above)
Comment (tylerlevine) [reply package.json]: `npm audit` is correct, I'm not sure why a script is being added though. It is not necessary.

File: .drone.jsonnet (hunk 3 above)
Comment (james8nguyen) [reply .drone.jsonnet]: It is close to "ignore error" and do not stop the run or fail entire pipeline.

File: package.json (hunk 1 above)
Comment (james8nguyen) [reply package.json]: Correct. Technically, I did not have to add it as a script in package.json; however visually, I added so that we could see "audit" being supported. 😄

Comment (james8nguyen) [reply package.json]: In the script, "npm audit" is correct. When calling externally, the typed command should "npm run audit".

File: .eslintrc.json (hunk 2 above)
Comment (james8nguyen) [reply .eslintrc.json]: Sure but what do you have in mind here?
I was thinking of implementing multiple tests to have a baseline template, although there are still failing tests.

**Comment by james8nguyen** [comment]
Given the latest requirements of obsoleting support for node 6, adding support for node 10, and extracting lint test into a different PR, I'm closing this PR and consolidating into PR https://github.com/BitGo/key-recovery-service-v2/pull/70. Thank you all for your valuable feedbacks.
